packages/ui/src/components/va-progress-bar/VaProgressBar.vue

```diff
@@ -1,112 +1,96 @@
-      <template v-else>
+    <div class="va-progress-bar__wrapper">
+      <div class="va-progress-bar__buffer" :style="bufferStyle" />
+
+      <template v-if="indeterminate">
         <div
-          :style="{backgroundColor: colorComputed, animationDirection: reverse ? 'reverse' : 'normal'}"
-          class="va-progress-bar__overlay__indeterminate-start"
+          :style="intermediateStyle"
+          class="va-progress-bar__progress--indeterminate-start"
         />
         <div
-          :style="{backgroundColor: colorComputed, animationDirection: reverse ? 'reverse' : 'normal'}"
-          class="va-progress-bar__overlay__indeterminate-end"
+          :style="intermediateStyle"
+          class="va-progress-bar__progress--indeterminate-end"
         />
       </template>
+      <div v-else class="va-progress-bar__progress" :style="progressStyle">
+        <slot v-if="isLarge" />
+      </div>
     </div>
   </div>
 </template>
 
 <script lang="ts">
-import { Options, mixins, prop, Vue } from 'vue-class-component'
-
-import { normalizeValue } from '../../services/utils'
-import { ProgressComponentMixin } from './ProgressComponentMixin'
-import ColorMixin from '../../services/color-config/ColorMixin'
-import { SizeMixin } from '../../mixins/SizeMixin'
-
-class ProgressBarProps {
-  buffer = prop<number>({ type: Number, default: 100 })
-  rounded = prop<boolean>({ type: Boolean, default: true })
-  size = prop<string | number>({ type: [Number, String], default: 'medium' })
-  reverse = prop<boolean>({ type: Boolean, default: false })
-  color = prop<string>({ type: String, default: 'primary' })
-}
+import clamp from 'lodash/clamp'
+import { computed, defineComponent, PropType } from 'vue'
+import { useColors } from '../../services/color-config/color-config'
 
-const ProgressBarPropsMixin = Vue.with(ProgressBarProps)
+type VaProgressSize = 'medium' | 'large' | 'small';
 
-@Options({
+export default defineComponent({
   name: 'VaProgressBar',
-})
-export default class VaProgressBar extends mixins(
-  ProgressComponentMixin,
-  ColorMixin,
-  SizeMixin,
-  ProgressBarPropsMixin,
-) {
-  get large () {
-    return this.size === 'large'
-  }
-
-  get small () {
-    return this.size === 'small'
-  }
 
-  get normalizedBuffer () {
-    if (this.indeterminate) {
-      return 100
+  props: {
+    modelValue: { type: Number, default: 0 },
+    indeterminate: { type: Boolean, default: false },
+    color: { type: String, default: 'primary' },
+    size: {
+      type: [Number, String] as PropType<number | VaProgressSize>,
+      default: 'medium',
+    },
+    buffer: { type: Number, default: 100 },
+    rounded: { type: Boolean, default: true },
+    reverse: { type: Boolean, default: false },
+  },
+
+  setup (props) {
+    const { getColor } = useColors()
+
+    const isLarge = computed(() => props.size === 'large')
+    const isTextSize = computed(() => typeof props.size === 'string' && ['small', 'medium', 'large'].includes(props.size))
```

Comment:
```suggestion
    const isTextSize = computed(() => typeof props.size === 'string' && ['small', 'medium', 'large'].includes(props.size))
```
 const isTextSize = computed(() => props.size instanceof VaProgressSize

Reply:
Instanceof checks prototype constructors and can not be used on abstract typescript type.

Reply:
Oh, my bad